src/sentry/search/django/backend.py

```diff
@@ -8,286 +8,445 @@
-                GroupStatus.PENDING_MERGE,
-            )
-            queryset = queryset.exclude(status__in=status_in)
-        else:
-            queryset = queryset.filter(status=status)
+    def build(self, queryset, parameters):
+        for name, condition in self.conditions.items():
+            if name in parameters:
+                queryset = condition.apply(queryset, name, parameters)
+        return queryset
 
-        if bookmarked_by:
-            queryset = queryset.filter(
-                bookmark_set__project=project,
-                bookmark_set__user=bookmarked_by,
-            )
 
-        if assigned_to:
-            teams = []
-            try:
-                member = OrganizationMember.objects.get(
-                    user=assigned_to,
-                    organization_id=project.organization_id,
-                )
-            except OrganizationMember.DoesNotExist:
-                pass
-            else:
-                teams = member.get_teams()
-
-            queryset = queryset.filter(
-                Q(assignee_set__user=assigned_to, assignee_set__project=project) |
-                Q(assignee_set__team__in=teams)
-            )
-        elif unassigned in (True, False):
-            queryset = queryset.filter(
-                assignee_set__isnull=unassigned,
-            )
+class Condition(object):
+    """\
+    Adds a single filter to a ``QuerySet`` object. Used with
+    ``QuerySetBuilder``.
+    """
+
+    def apply(self, queryset, name, parameters):
+        raise NotImplementedError
+
+
+class CallbackCondition(Condition):
+    def __init__(self, callback, destructive=True):
+        self.callback = callback
+        self.destructive = destructive
+
+    def apply(self, queryset, name, parameters):
+        accessor = parameters.pop if self.destructive else parameters.__getitem__
+        return self.callback(queryset, accessor(name))
+
+
+class ScalarCondition(Condition):
+    """\
+    Adds a scalar filter (less than or greater than are supported) to a
+    ``QuerySet`` object. Whether or not the filter is inclusive is defined by
+    the '{parameter_name}_inclusive' parameter.
+    """
+
+    def __init__(self, field, operator, destructive=True):
+        assert operator in ['lt', 'gt']
+        self.field = field
+        self.operator = operator
+        self.destructive = destructive
+
+    def apply(self, queryset, name, parameters):
+        accessor = parameters.pop if self.destructive else parameters.__getitem__
+        inclusive = accessor('{}_inclusive'.format(name))
+        return queryset.filter(**{
+            '{}__{}{}'.format(
+                self.field,
+                self.operator,
+                'e' if inclusive else ''
+            ): accessor(name)
+        })
+
+
+def get_sql_table(model):
+    return '{}'.format(model._meta.db_table)
+
+
+def get_sql_column(model, field):
+    "Convert a model class and field name to it's (unquoted!) SQL column representation."
```

Comment:
This is MySQL's fault. This method never takes user input, though. It's used for figuring out column names to construct manual joins or `WHERE` conditions.